custom/m4change/reports/__init__.py

```diff
@@ -13,18 +13,23 @@ def _is_location_CCT(location):
 
 
 # A workaround to the problem occurring if a user selects 'All' in the first location filter dropdown
-def get_location_hierarchy_by_id(location_id, domain, CCT_only=False):
+def get_location_hierarchy_by_id(location_id, domain, user, CCT_only=False):
     if location_id is None or len(location_id) == 0:
-        return [location.get_id for location in Location.by_domain(domain) if not CCT_only or _is_location_CCT(location)]
+        return [
+            location.location_id for location in SQLLocation.objects.accessible_to_user(domain, user)
+            if not CCT_only or _is_location_CCT(location)
+        ]
     else:
-        user_location = Location.get(location_id)
-        locations = [location.get_id for location in user_location.descendants if not CCT_only or _is_location_CCT(location)]
-        if not CCT_only or _is_location_CCT(user_location):
-            locations.insert(0, user_location.get_id)
-        return locations
+        user_location = SQLLocation.objects.get(location_id=location_id, domain=domain)
+        return [
+            location.get_id
+            for location in user_location.get_descendants(include_self=True).accessible_to_user(domain, user)
+            if not CCT_only or _is_location_CCT(location)
+        ]
```

Comment:
This looks safe, but it's worth noting that it may return an empty list, particularly if `location_id` isn't accessible to the user.